coda/coda_mdstore/tests/test_resourcesync.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+
+from django.contrib.sitemaps import Sitemap
+
+from .. import factories
+from .. import resourcesync
+
+pytestmark = [
+    pytest.mark.urls('coda_mdstore.urls'),
+    pytest.mark.django_db()
+]
+
+
+class FakeSitemap(Sitemap):
+    """Fake Sitemap object similar to resourcesync.BaseSitemap.
+
+    This will build but not persist model instances for testing
+    purposes.
+    """
+    lastmod = None
+    protocol = 'http'
+
+    def items(self):
+        # A miniumum of 50001 Bag objects are required to get multiple
+        # resource locations from the index view.
+        return factories.FullBagFactory.build_batch(5001)
+
+    def location(self, obj):
+        return "/bag/%s" % obj.name
+
+
+def test_index_context(rf):
+    """Test the context data in the response returned from `index`.
+
+    `index` is essentially the function defined in
+    django.contrib.sitemaps.views.index, but it alters the data included
+    in the response context slightly. This tests that the data is in the
+    correct form.
+    """
+    sitemaps = {'001': FakeSitemap}
+    request = rf.get('/')
+    response = resourcesync.index(request, sitemaps, 'mdstore/resourceindex.xml')
+
+    # Expected filenames. The full location would look something like
+    # http://example.com/resourcelist-001.xml
+    resource_list_1 = 'resourcelist-001.xml'
+    resource_list_2 = 'resourcelist-002.xml'
+
+    locations = response.context_data['sitemaps']
+
+    assert resource_list_1 in locations[0]
+    assert resource_list_2 in locations[1]
+
+
+def test_sitemap_content(rf):
+    """Test the content in the response returned from `sitemap`.
+
+    `sitemap` is essentially the function defined in
+    django.contrib.sitemaps.views.sitemap, but it alters the data included
+    in the response context slightly. This tests that the data is present
+    in the content.
+    """
+    bags = factories.FullBagFactory.create_batch(10)
+
+    request = rf.get('/')
+    response = resourcesync.sitemap(request, resourcesync.sitemaps, 1, 'mdstore/sitemap.xml')
+    response.render()
+
+    # Verify that that the following urls are present for each bag in the
+    # bags batch.
+    for bag in bags:
+        assert 'http://example.com/bag/{0}'.format(bag.name) in response.content
+        assert 'http://example.com/bag/{0}.urls'.format(bag.name) in response.content
```

Comment:
These two assertions seem problematic. The first assertion is looking for essentially a prefix of the second. So both assertions will always be true if the second string is present.